### DISCUSSION TITLE
How to debug ?

### DISCUSSION THREAD

**Discussion by etienne-sf** [enisdenjo/graphql-ws#250] [Q&A] [answered]
Hello,

I'm working on a GraphQL plugin for Java ([graphql-java-generator](https://github.com/graphql-java-generator/graphql-maven-plugin-project)), that (should) implement graphql-transport-ws.

I'm trying the sample you gave, and I have a (apparently) working server from your sample. At least, it listen on port 4000 for something...
:)

When I try to access to it with the Java code generated , it happens ... just nothing.

From the issue #27, I see that you provided a solution, but I can't use it.
I've try [this scipt](https://gist.github.com/etienne-sf/17c4013beaf94b2a3569d652eed24743), but it complains with this message `Unexpected token ':'` on line 37 (`}: ServerOptions) {`).

_Isn't there a simple config option to activate log, and see what's happening ?_

In Java, I'm used to configure logging level, that allows to see what's happening in the used libraries.
And from what's written in issue #27, you've released something for that. 

But I have no clue on how to use it...

Etienne

**Comment by etienne-sf** [comment]
And if I remove the offending `: ServerOptions) {`, I'l stuck with this error:

`Cannot find package 'graphql-transport-ws'`

**Comment by enisdenjo** [comment]
I have a hard time understanding you. How are you using this package, what was your install process? Can you share some example code?

For implementing GraphQL over WebSocket supported by this library, please consult [the Protocol document](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md).

**Comment by etienne-sf** [comment]
Yes, right.
I have to say that everything is crystal clear for me. But after re-reading it, it's not that obvious...
:(


I've read the document you pointed out: it's very clear, and I implemented it in Java, in both the client and the server side.
It works well between them.

Now I want to do integration tests against your implementation, as it is the reference implementation. 

**The issue I have is that nothing happens when my Java client calls your server.**

So I need to have some information on what's happening on server side, that is: some sort of debug logging

I've tried what you've explained in the issue #27, but I was unsuccessful in using the sample code you provided in issue #27. This code I've done is [this script](https://gist.github.com/etienne-sf/17c4013beaf94b2a3569d652eed24743). But it doesn't work: it complains with this message `Unexpected token ':'` on line 37 (`}: ServerOptions) {`).

My question is:
_Is there any way to get server debug output from your library ?_

Etienne

**Comment by enisdenjo** [comment, marked as answer]
> The issue I have is that nothing happens when my Java client calls your server.
> 
> So I need to have some information on what's happening on server side, that is: some sort of debug logging

From my understanding, the `server.js` does not even start, nothing _can_ happen.

> I've tried what you've explained in the issue #27, but I was unsuccessful in using the sample code you provided in issue #27. This code I've done is this script. But it doesn't work: it complains with this message Unexpected token ':' on line 37 (}: ServerOptions) {).

Looking at your script briefly, I found 2 issues:
-  You are referencing an old issue while the lib was named differently `graphql-transport-ws`. Since then, **[it has been deprecated in favor of `graphql-ws` and should not be used](https://www.npmjs.com/package/graphql-transport-ws)**.
  I've fixed the example from #27 to point to the ["ws server usage with console logging" recipe](https://github.com/enisdenjo/graphql-ws#logging), please consult it instead.
- Line 37 contains TypeScript code (`}: ServerOptions`). You can not run TypeScript with Node. [TypeScript is a superset of JavaScript](https://www.typescriptlang.org/) **that transpiles to JavaScript**. Please consult the ["ws server usage with console logging" recipe](https://github.com/enisdenjo/graphql-ws#logging) for a JavaScript example.

**Reply by etienne-sf** [reply]
Thank you for your support. 

I have some journey, learning JavaScript, node...

When I succeed in that, and with the info you gave, I guess I'll be right. 

I have only little time on this subject, so it will take some days for me to manage all this. 


Thanks again for your very reactive support
:)

Étienne
